src/nvim/ex_docmd.c

```diff
@@ -9952,3 +9953,104 @@ bool cmd_can_preview(char_u *cmd)
 
   return false;
 }
+
+/// Return the list of the commands in the specified buffer,
+/// if no buffer is specified, then return list of global(non-buffer) the
+/// commands
+///
+/// @param[in] buf the specified buffer, if NULL then global commands required
+///
+/// @return The array of the commands and the information of each command
+ArrayOf(Dictionary) get_commands(buf_T *buf)
```

Comment:
Not sure if it's the best convention, but our current convention is to name this `commands_array`. 

Existing examples: `mode_style_array`, `keymap_array`. Until a better convention is found, we should always do the same thing, to avoid having numerous inconsistent patterns.

Reply:
OK got it